include/irr/core/math/aabb2dSIMDIntBase.h

```diff
@@ -0,0 +1,259 @@
+#include "irr/core/memory/memory.h"
+#include "irr/core/alloc/AlignedBase.h"
+#include "../include/vectorSIMD.h"
+
+#include <iostream>
+
+#define DWN_CAST_THIS_PTR static_cast<const CRTP<VECTOR_TYPE>*>(this)
+#define SIGN_FLIP_MASK_XY	core::vector3du32_SIMD(0x80000000, 0x80000000, 0x00000000, 0x00000000)
+
+namespace irr { namespace core {
+
+
+template<template<typename> typename CRTP, typename VECTOR_TYPE>
+class aabbSIMDBase : public AlignedBase<_IRR_VECTOR_ALIGNMENT>
+{
+	static_assert(
+		std::is_same<VECTOR_TYPE, core::vector2di32_SIMD>::value ||
+		std::is_same<VECTOR_TYPE, core::vector2du32_SIMD>::value ||
+		std::is_same<VECTOR_TYPE, core::vector2df_SIMD>::value,
+		"assertion failed: cannot use this type");
+
+public:
+	inline void addPoint(const VECTOR_TYPE& other)
+	{
+		DWN_CAST_THIS_PTR->addBox();
+	}
+
+	inline VECTOR_TYPE getExtent() const
+	{
+		return DWN_CAST_THIS_PTR->getMaxEdge() - static_cast<const CRTP<VECTOR_TYPE>*>(this)->getMinEdge();
+	}
+
+	//need operator/ for core::vector32_i<uint32_t> and core::vector32_i<uint32_t>
+	/*inline VECTOR_TYPE getCenter() const
+	{
+		return static_cast<const CRTP<VECTOR_TYPE>*>(this)->getMaxEdge() - static_cast<const CRTP<VECTOR_TYPE>*>(this)->getMinEdge();
+	}*/
+
+	//! Resets the bounding box to a one-point box.
+	/** \param x X coord of the point.
+	\param y Y coord of the point. */
+	inline void reset(const VECTOR_TYPE& point)
+	{
+		CRTP<VECTOR_TYPE> pointBox(point.xyxy());
+		internalBoxRepresentation= pointBox;
+	}
+
+	//! Repairs the box.
+	/** Necessary if for example MinEdge and MaxEdge are swapped. */
+	inline CRTP<VECTOR_TYPE> repair()
+	{
+		//
+		//but will it work for aabbSIMDf for sure?
+		//
+		//
+		reset(DWN_CAST_THIS_PTR->getMinEdge());
+		addPoint(DWN_CAST_THIS_PTR->getMaxEdge());
+
+		return *this;
+	}
+
+	//! Check if the box is empty.
+	/** This means that there is no space between the min and max edge.
+	\return True if box is empty, else false. */
+	inline bool isEmpty() const
+	{
+		return return (getMinEdge() == getMaxEdge()).all();;
+	}
+
+	//! Get the surface area of the box in squared units
+	inline float getArea() const
+	{
+		VECTOR_TYPE a = DWN_CAST_THIS_PTR->internalBoxRepresentation.getMaxEdge().x - DWN_CAST_THIS_PTR->internalBoxRepresentation.getMinEdge().x;
+		VECTOR_TYPE b = DWN_CAST_THIS_PTR->internalBoxRepresentation.getMaxEdge().y - DWN_CAST_THIS_PTR->internalBoxRepresentation.getMinEdge().y;
+		return a * b;
+	}
+
+	//! Get center of the bounding box
+	/** \return Center of the bounding box. */
+	inline core::vectorSIMDf getCenter() const
+	{
+		return (internalBoxRepresentation.zwzw() - internalBoxRepresentation) / VECTOR_TYPE(2);
+	}
+
+	//! Determines if the axis-aligned box intersects with another axis-aligned box.
+	/** \param other: Other box to check a intersection with.
+	\return True if there is an intersection with the other box,
+	otherwise false. */
+	bool intersectsWithBox(const CRTP<VECTOR_TYPE>& other) const
+	{
+		return
```

Comment:
remember about the stack overflow algorithm

Reply:
sure